### PR TITLE
serializable RemoteAssetNodes

### PR DESCRIPTION
**Pull request by alangenfeld**
Make the global and repository scoped `RemoteAssetNode` serializable so we can persist/cache the resolved global asset node of a workspace.

## How I Tested These Changes

added test

### CODE REVIEW

**Comment by alangenfeld** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24827?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24827** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24827?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24824** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24824?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24822** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24822?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24820** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24820?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py [hunk 1]
```diff
@@ -48,23 +52,28 @@
     from dagster._core.selector.subset_selector import DependencyGraph
 
 
+@whitelist_for_serdes
+@record
+class SelectAssetNode:
+    selector: RepositorySelector
+    asset: Annotated[
+        "ExternalAssetNode", ImportFrom("dagster._core.remote_representation.external_data")
+    ]
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py:61]: was having a hard time naming this thing - the named variant of (location, asset) tuple 
"scoped" was another idea

**Review by alangenfeld — COMMENTED** [review]
I don't intend to land this until i can validate that it asset node thing that we cache allows us to load pages faster. I think we may need to add in the job snapshot id so we can look up job / op details without the whole workspace/repo. 

Just wanted to get eyes on this early to get naming and data layout right since they will be come much higher friction to change once this becomes serializable.

File: python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py (hunk 1 above)
Comment (OwenKephart) [reply python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py:61]: I like `RepositoryScopedAssetNode` -- it's possible that we actually want this to be generic though? i.e. `RepositoryScopedEntityNode[T_EntityNode]`

**Review by OwenKephart — APPROVED** [review]
Just logging here that we may want to do something similar for asset check nodes in the future (although this is probably much less critical).

But it seems like we'd still want two separate serializable classes in that case, so seems ok

**Pull request merged**